From de4ebba0ba745fadbc0699c027e29e07ef7c4de1 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Thu, 17 Dec 2009 16:37:24 +0200
Subject: [PATCH] xonix.c: Unused vars and other compiling fixes


Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 xonix.c |   14 ++++++++------
 1 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/xonix.c b/xonix.c
index 87983ab..1f73a80 100644
--- a/xonix.c
+++ b/xonix.c
@@ -83,6 +83,10 @@ int 				gLoops;					/* Schleifenzaehler im Animate	*/
 /* Forward-Declarations.                                                    */
 /* ------------------------------------------------------------------------ */
 
+extern int  DrawCompleteInside(void); /* x11.c */
+extern int  DisplayHighScore(void);   /* x11.c */
+extern void ScorePoints (int points); /* x11.c */
+
 void Do_New (void);
 void NewFlyer (void);
 void NewRunner (void);
@@ -739,8 +743,8 @@ Boolean	SeedFillUp (int xStart, int yStart, Boolean wayToFill)
 	segStack *fillStack;
 	mySegment	sg;
 	Boolean runFlag;
-	int l, x1, x2, dy;
-	unsigned char ov, av, nv, wv;
+	int l;
+	unsigned char ov, av=0, nv, wv;
 	int i, j, m, n;
 
 	if ((fillStack = InitSegmentStack ()) == NIL_POINTER)
@@ -911,7 +915,6 @@ skip:		for (xStart += RATIO; xStart <= sg.xr &&
 
 unsigned char NewRunnerPosition (void)
 {
-    int i, j;
     Boolean     bounced;
     unsigned char bouncePartner;
 
@@ -1041,7 +1044,6 @@ void FillNewArea (void)
 	int x1, y1, x2, y2;
 	int percent, oldFillCount;
 	Boolean	wayToFill;
-	double bonusFactor;
 	
 	GWorldEntry ();
 	
@@ -1094,7 +1096,7 @@ void FillNewArea (void)
 
 Boolean NewEaterPosition (void)
 {
-	int	i, j;
+	int	i;
 	int nextX, nextY;
 	Boolean hbFlag, vbFlag, returnFlag;
 	unsigned char bouncePartner;
@@ -1310,7 +1312,7 @@ Boolean VerticalBounceCheck (int x, int y, int size,
     
 Boolean NewFlyerPosition (void)
 {
-    int i, j;
+    int i;
     unsigned char   bouncePartner;
     Boolean     returnFlag, vbFlag, hbFlag;
 
-- 
1.6.5

