From e9cb4c619382d2cb92029a3eb971f55b7a2086b3 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Thu, 17 Dec 2009 17:36:59 +0200
Subject: [PATCH] Xonix.ad: Change to neutral messages

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 Xonix.ad |   10 +++++-----
 1 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Xonix.ad b/Xonix.ad
index 673bb71..12a1448 100644
--- a/Xonix.ad
+++ b/Xonix.ad
@@ -50,7 +50,7 @@ Xonix.timeStep: 50
 *score_shell*Box.borderWidth:	0
 *score_shell*score_msg.label:	Best Xonixers:
 *score_shell*score_headl.label:	Score Level    Date    User ID    Full name
-*score_shell*score_done.label:	Great!
+*score_shell*score_done.label:	Ok
 *score_shell*score_done.background: white
 
 !
@@ -83,24 +83,24 @@ Xonix.timeStep: 50
 !
 ! the game over dialog
 !
-*gameover_shell.title:		Pity!
+*gameover_shell.title:		Game over
 *gameover_box.background:	khaki
 *gameover_box.orientation:	vertical
 *gameover_buttonbox.orientation: horizontal
 *gameover_buttonbox.background:	khaki
 *gameover_buttonbox.borderWidth: 0
-*gameover_msg.label:		Game over, dude!
+*gameover_msg.label:		Game over
 *gameover_msg.font:	-*-new century schoolbook-bold-i-*-*-*-180-*-*-*-*-*-*
 *gameover_msg.borderWidth:	0
 *gameover_msg.foreground:	sienna
 *gameover_msg.background:	khaki
-*gameover_iknow.label:		I know :-(
+*gameover_iknow.label:		Exit
 *gameover_iknow.foreground:	light yellow
 *gameover_iknow.background:	dark orchid
 *gameover_iknow.accelerators:	#augment \
 				<Key>Q: set() notify() unset() \n\
 				<Key>Escape: set() notify() unset()
-*gameover_goon.label:		Once more! :)
+*gameover_goon.label:		Play again
 *gameover_goon.foreground:	light yellow
 *gameover_goon.background:	dark orchid
 *gameover_goon.accelerators:	#augment \
-- 
1.6.5

